types/styled-components/test/index.tsx

```diff
@@ -25,6 +25,67 @@ import {} from "styled-components/cssprop";
  * general usage
  */
 
+interface AllOptionalProps {
+    text?: boolean;
+    altText?: boolean;
+}
+
+const AllOptionalPropComponent: React.SFC<AllOptionalProps> =
+    ({ text, altText, children }) => (
+        <div>
+            {text || altText || 'Hello World'}
+            {children}
+        </div>
+    );
+
+const StyledAllOptionalPropComponent = styled(AllOptionalPropComponent)`
+    max-width: 100px;
+`;
+
+// Check that passing no props to a component with only optional props works.
+// If we aren't explicitly setting a `children` prop, then this would produce
+// a "no properties in common" error.
+const UseStyledAllOptionalPropComponent: React.SFC = () => (
+    <StyledAllOptionalPropComponent>
+        blah
```

Comment:
There's a bug in TS that doesn't actually trigger excess property errors with `children` in certain cases, so this would also need an `$ExpectError` equivalent.

Don't define such a component with `React.SFC`, use a function declaration as it's the only thing not directly poisoned by the automatic `{ children?: React.ReactNode }` which must be killed.

Reply:
Actually, since you did declare `AllOptionalPropComponent` with `React.SFC`, it _should_ have an implicit `{ children?: React.ReactNode }` in it. I wonder if the `&` in the `React.SFC` is erasing it from the extracted types in `React.ComponentProps`.

Yet one more reason why the automatic `{ children?: React.ReactNode }` must be killed.